keepalived insists on creating PID files, so at least don't let it create in
/run (to avoid clash with OpenRC), but /run/keepalived. keepalived will
automatically create and delete this directory.

This is a downstream patch.

--- a/keepalived/core/main.c
+++ b/keepalived/core/main.c
@@ -87,7 +87,7 @@
 #endif
 static char *syslog_ident;				/* syslog ident if not default */
 char *instance_name;					/* keepalived instance name */
-bool use_pid_dir;					/* Put pid files in /var/run/keepalived or @localstatedir@/run/keepalived */
+bool use_pid_dir = true;				/* Put pid files in /var/run/keepalived or @localstatedir@/run/keepalived */
 
 unsigned os_major;					/* Kernel version */
 unsigned os_minor;
